lua/nvim-web-devicons.lua

```diff
@@ -1712,13 +1736,35 @@ local function setup(opts)
     global_opts.default = true
   end
 
+  if user_icons.strict then
+    global_opts.strict = true
+  end
+
   global_opts.color_icons = if_nil(user_icons.color_icons, global_opts.color_icons)
 
   if user_icons.override and user_icons.override.default_icon then
     default_icon = user_icons.override.default_icon
   end
 
-  icons = vim.tbl_extend("force", icons, user_icons.override or {})
+  local user_filename_icons = user_icons.override and user_icons.override.by_filename
```

Comment:
Let's change the setup schema to
```lua
{
  override = {
    ---
  },
  override_by_filename = {
    ---
  },
  override_by_extension = {
    ---
  },
---
}
```

Having the specific overrides next to the icons is quite confusing.